src/Profile.cpp

```diff
@@ -134,212 +91,74 @@ namespace geopm
         ++m_overhead_time_shutdown;
         --m_overhead_time_shutdown;
 #endif
-        std::string sample_key(m_key_base + "-sample");
-        int shm_num_rank = 0;
-
-        init_prof_comm(std::move(m_comm), shm_num_rank);
-        std::string step;
+        init_cpu_set(m_num_cpu);
         try {
-            step = "ctl_msg";
-            init_ctl_msg(sample_key);
-            step = "cpu_set";
-            init_cpu_set(m_num_cpu);
-            step = "cpu_affinity";
-            init_cpu_affinity(shm_num_rank);
-            step = "table";
-            init_table(sample_key);
-            step = "app_status";
             init_app_status();
-            step = "app_record_log";
             init_app_record_log();
-            m_is_enabled = true;
         }
         catch (const Exception &ex) {
-            std::cerr << "Warning: <geopm> Controller handshake failed at step "
-                      << step << ", running without geopm." << std::endl;
+            std::cerr << "Warning: <geopm> Failed to connect with geopmd, running without geopm." << std::endl;
             int err = ex.err_value();
-            if (err != GEOPM_ERROR_RUNTIME) {
-                char tmp_msg[PATH_MAX];
-                geopm_error_message(err, tmp_msg, sizeof(tmp_msg));
-                tmp_msg[PATH_MAX-1] = '\0';
-                std::cerr << tmp_msg << std::endl;
-            }
-            m_is_enabled = false;
+            char tmp_msg[PATH_MAX];
+            geopm_error_message(err, tmp_msg, sizeof(tmp_msg));
+            tmp_msg[PATH_MAX-1] = '\0';
+            std::cerr << tmp_msg << std::endl;
         }
+        m_is_enabled = true;
```

Comment:
Did you intend to make `m_is_enabled = true` even in the case where the prior init steps throw an exception?